/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
10_Y-MPP4_2.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
13_A-HSC_1.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
14_A-MPP1_1.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
15_A-MPP2_1.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
16_A-MPP3_1_new.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
17_A-MPP4_1_new.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
18_A-HSC_2_new.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
19_A-MPP1_2_new.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
1_Y-HSC_1.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
20_A-MPP2_2_new.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
21_A-MPP3_2_new.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
22_A-MPP4_2_new.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
25_Y-GMP_1.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
26_Y-MEP_1.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
27_Y-CLP_1.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
28_A-GMP_1.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
29_A-MEP_1.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
2_Y-MPP1_1.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
30_A-CLP_1.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
31_Y-GMP_2.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
32_Y-MEP_2_new.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
33_Y-CLP_2_new.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
34_A-GMP_2.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
35_A-MEP_2_new.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
36_A-CLP_2_new.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
3_Y-MPP2_1.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
4_Y-MPP3_1.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
5_Y-MPP4_1.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
6_Y-HSC_2.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
7_Y-MPP1_2.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
8_Y-MPP2_2.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
9_Y-MPP3_2.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
O-HSC_1.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
O-HSC_2.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
Y-HSC_1.fastq.gz &
/root/myPrograms/TrimGalore-0.4.5/trim_galore --illumina -q 20 --fastqc -o /root/taka/ATAC-seq/fastq/fastq_trim \
Y-HSC_2.fastq.gz &
##############################################
ls -1 *gz|perl -pe 's/(.+)\n/STAR --genomeDir \/root\/resources\/mm10_noanno\/ \\
--readFilesCommand zcat \\
--runThreadN 35 \\
--alignIntronMax 1 \\
--outFilterMismatchNoverLmax 0.09 \\
--alignMatesGapMax 2000 \\
--outFilterMultimapNmax 1 \\
--alignEndsType EndToEnd \\
--readFilesIn \/root\/taka\/ATAC-seq\/fastq\/fastq_trim\/$1 \\
--outSAMtype BAM SortedByCoordinate \\
--outFileNamePrefix $1_
######\n/g'

ls -1 *gz|perl -pe 's/(.+)\n/STAR --genomeDir \/root\/resources\/mm10_noanno\/ \\
--readFilesCommand zcat \\
--runThreadN 35 \\
--alignIntronMax 1 \\
--outFilterMismatchNoverLmax 0.09 \\
--alignMatesGapMax 2000 \\
--outFilterMultimapNmax 1 \\
--alignEndsType EndToEnd \\
--readFilesIn \/root\/taka\/ATAC-seq\/fastq\/fastq_trim\/$1 \\
--outSAMtype BAM SortedByCoordinate \\
--outFileNamePrefix $1_
######\n/g'

STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/10_Y-MPP4_2_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 10_Y-MPP4_2_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/13_A-HSC_1_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 13_A-HSC_1_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/14_A-MPP1_1_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 14_A-MPP1_1_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/15_A-MPP2_1_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 15_A-MPP2_1_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/16_A-MPP3_1_new_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 16_A-MPP3_1_new_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/17_A-MPP4_1_new_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 17_A-MPP4_1_new_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/18_A-HSC_2_new_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 18_A-HSC_2_new_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/19_A-MPP1_2_new_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 19_A-MPP1_2_new_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/1_Y-HSC_1_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 1_Y-HSC_1_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/20_A-MPP2_2_new_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 20_A-MPP2_2_new_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/21_A-MPP3_2_new_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 21_A-MPP3_2_new_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/22_A-MPP4_2_new_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 22_A-MPP4_2_new_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/25_Y-GMP_1_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 25_Y-GMP_1_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/26_Y-MEP_1_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 26_Y-MEP_1_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/27_Y-CLP_1_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 27_Y-CLP_1_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/28_A-GMP_1_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 28_A-GMP_1_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/29_A-MEP_1_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 29_A-MEP_1_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/2_Y-MPP1_1_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 2_Y-MPP1_1_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/30_A-CLP_1_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 30_A-CLP_1_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/31_Y-GMP_2_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 31_Y-GMP_2_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/32_Y-MEP_2_new_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 32_Y-MEP_2_new_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/33_Y-CLP_2_new_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 33_Y-CLP_2_new_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/34_A-GMP_2_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 34_A-GMP_2_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/35_A-MEP_2_new_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 35_A-MEP_2_new_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/36_A-CLP_2_new_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 36_A-CLP_2_new_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/3_Y-MPP2_1_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 3_Y-MPP2_1_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/4_Y-MPP3_1_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 4_Y-MPP3_1_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/5_Y-MPP4_1_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 5_Y-MPP4_1_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/6_Y-HSC_2_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 6_Y-HSC_2_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/7_Y-MPP1_2_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 7_Y-MPP1_2_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/8_Y-MPP2_2_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 8_Y-MPP2_2_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/9_Y-MPP3_2_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix 9_Y-MPP3_2_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/O-HSC_1_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix O-HSC_1_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/O-HSC_2_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix O-HSC_2_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/Y-HSC_1_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix Y-HSC_1_trimmed.fq.gz_
######
STAR --genomeDir /root/resources/mm10_noanno/ \
--readFilesCommand zcat \
--runThreadN 35 \
--alignIntronMax 1 \
--outFilterMismatchNoverLmax 0.09 \
--alignMatesGapMax 2000 \
--outFilterMultimapNmax 1 \
--alignEndsType EndToEnd \
--readFilesIn /root/taka/ATAC-seq/fastq/fastq_trim/Y-HSC_2_trimmed.fq.gz \
--outSAMtype BAM SortedByCoordinate \
--outFileNamePrefix Y-HSC_2_trimmed.fq.gz_
######

